### PR TITLE
TE-10481 Removed wrong message when bridge is stricter than parent class

### PR DESCRIPTION
**Pull request by geega**
- Developer(s): @geega

- Ticket: https://spryker.atlassian.net/browse/TE-10481

- Docs PR: ACADEMY_URL_HERE

- Release Group: https://release.spryker.com/release-groups/view/3930

- merge: merge

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   ArchitectureSniffer              | minor                 |                       |

-----------------------------------------

##### Change log

Improvements

- Adjusted `BridgeMethodsRule` so it fixes the wrong message when the bridge is stricter than the parent class.

### CODE REVIEW

File: tests/_data/Common/Bridge/BridgeMethodsTest/testRuleDoesNotApplyWhenBridgeMethodsAreCorrect.php [hunk 1]
```diff
@@ -44,5 +44,5 @@ class DataObject
 
 interface CacheClientInterface
 {
-    public function save(string $key, DataObject $data, string $prefix = null);
-}
\ No newline at end of file
+    public function save(string $key, DataObject $data, $prefix = null, array $options = null);
```
Comment (dereuromark) [inline tests/_data/Common/Bridge/BridgeMethodsTest/testRuleDoesNotApplyWhenBridgeMethodsAreCorrect.php:47]: What is the parent interface signature here? Can you paste it please for easier review on those changes?

File: src/Common/Bridge/BridgeMethodsRule.php [hunk 2]
```diff
@@ -273,4 +283,22 @@ protected function getBridgedInterfaceReflection(array $classMethods): ?Reflecti
...
+     * @param array<\ReflectionParameter> $params
+     *
+     * @return int
+     */
+    protected function countRequireParams(array $params): int
+    {
+        $countParams = 0;
+
+        foreach ($params as $param) {
+            if (!$param->isDefaultValueAvailable()) {
```
Comment (dereuromark) [inline src/Common/Bridge/BridgeMethodsRule.php:297]: I dont think is is correct as per ticket.

File: src/Common/Bridge/BridgeMethodsRule.php [hunk 3]
```diff
@@ -203,7 +203,7 @@ protected function findNotMatchingMethodsForBridgeInterface(InterfaceNode $inter
             $interfaceMethodName = sprintf('%s::%s', $interfaceNode->getFullQualifiedName(), $interfaceMethod->getName());
             $interfaceMethodReflection = new ReflectionMethod($interfaceMethodName);
 
-            if ($this->compareTwoMethodsForBridgeInterface($interfaceMethodReflection, $bridgedInterfaceReflectionMethod)) {
+            if ($this->comporeBridgeAndParentMethodInterface($interfaceMethodReflection, $bridgedInterfaceReflectionMethod)) {
```
Comment (dereuromark) [inline src/Common/Bridge/BridgeMethodsRule.php:206]: compore?

File: src/Common/Bridge/BridgeMethodsRule.php (hunk 2 above)
Comment (dereuromark) [reply src/Common/Bridge/BridgeMethodsRule.php:297]: We can keep this, though. 
It seems fine to check for optional args and ignore those.

File: tests/_data/Common/Bridge/BridgeMethodsTest/testRuleDoesNotApplyWhenBridgeMethodsAreCorrect.php (hunk 1 above)
Comment (dereuromark) [reply tests/_data/Common/Bridge/BridgeMethodsTest/testRuleDoesNotApplyWhenBridgeMethodsAreCorrect.php:47]: We can also keep stricter param types for now, though.

**Comment by dereuromark** [comment]
ping @geega

**Comment by geega** [comment]
@dereuromark  Yes. Thanks

**Comment by dereuromark** [comment]
Always also remove branches!

**Pull request closed without merging**